### ISSUE TITLE
Improvements 

### ISSUE THREAD

**Issue by wangcj05** [idaholab/BayCal#3]
- Unfortunately the "compute Covariance" does not work.... it returns all NaN... I had to input the diagonal covariance by hand.
- RavenRunningRaven does not work as well because the output of the inner is not interpreted as an HistorySet but as pointSet (no matter what I do in the outer dataobject). I had to use the GenericCode interface for the inner 
- The <reduction> node if does not have the <truncationRank> simply silently "skips" the reduction (Without informing the user)
- If the <truncationRank> is inputted and == 0 (optimal rank should be computed atuomatically) the reduction is skipped because this line "if self._subspace is None and self._truncationRank:" is evaluated as False (since if "self._truncationRank is 0.0 and it is False)...It should be modified to "if self._subspace is None and self._truncationRank is not None"
- If the <batchSize> is set to > 1 the code just crashes after the first evaluation. 
- It would be nice to have the possibility to provide the <expTargets> values as CSV (or RAVEN data object) and maybe also to associate a pivot parameter to them (so the underlying model can be run in parallel (e.g. the inner RAVEN that runs relap5 for each experiement). Now I am forced to have the inner to run in serial to keep the right order of values (mapping with the values in <expTargets>)
